cypress/plugins/postMessageAs.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+require('@babel/polyfill');
+require('isomorphic-fetch');
+
+const {Client4} = require('mattermost-redux/client');
+const axios = require('axios');
+
+const cypressConfig = require('../../cypress.json');
+
+module.exports = async ({sender, message, channelId}) => {
+    const url = `${cypressConfig.baseUrl}/api/v4/users/login`;
+
+    const response = await axios({url, method: 'post', data: {login_id: sender.username, password: sender.password}});
+    const token = response.headers.token;
+
+    Client4.setUrl(cypressConfig.baseUrl);
+    Client4.setToken(token);
+
+    return Client4.createPost({
+        channel_id: channelId,
+        message,
+        type: '',
+    });
```

Comment:
I was thinking if this should be done in `cy.request` (wrapped in a new custom command) but then it'll loose the current session and will not be able to listen to Notification.  This turned out to be a better solution. 💯